tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -75,7 +167,76 @@ void mock_client(int writefd, int readfd)
     s2n_connection_free(conn);
     s2n_config_free(config);
 
-    /* Give the server a chance to a void a sigpipe */
+    /* Give the server a chance to avoid sigpipe */
+    sleep(1);
+
+    /* Session resumption */
+    conn = s2n_connection_new(S2N_CLIENT);
+    config = s2n_config_new();
+    s2n_config_disable_x509_verification(config);
+    s2n_connection_set_config(conn, config);
+    conn->client_protocol_version = S2N_TLS12;
+
+    s2n_connection_set_read_fd(conn, readfd);
+    s2n_connection_set_write_fd(conn, writefd);
+
+    /* Set session state on client connection */
+    if (s2n_connection_set_session(conn, serialized_session_state, serialized_session_state_length) < 0) {
+        result = 5;
+    }
+
+    if (s2n_negotiate(conn, &blocked) != 0) {
+        result = 6;
+    }
+
+    /* Make sure we did a abbreviated handshake */
+    if (!IS_RESUMPTION_HANDSHAKE(conn->handshake.handshake_type)) {
+        result = 7;
+    }
+
+    if (s2n_send(conn, MSG, sizeof(MSG), &blocked) != sizeof(MSG)) {
+        result = 8;
+    }
+
+    shutdown_rc = -1;
+    while(shutdown_rc != 0) {
+        shutdown_rc = s2n_shutdown(conn, &blocked);
+    }
+
+    s2n_connection_free(conn);
+    s2n_config_free(config);
+
+    /* Give the server a chance to avoid sigpipe */
+    sleep(1);
+
+    /* Session resumption with bad session state */
+    conn = s2n_connection_new(S2N_CLIENT);
+    config = s2n_config_new();
+    s2n_config_disable_x509_verification(config);
+    s2n_connection_set_config(conn, config);
+    conn->client_protocol_version = S2N_TLS12;
+
+    s2n_connection_set_read_fd(conn, readfd);
+    s2n_connection_set_write_fd(conn, writefd);
+
+    /* Change the protocol version in session state */
+    serialized_session_state[35] = 30;
```

Comment:
Should we ensure that 35 is within serialized_session_state_length limit?

Reply:
I think having a check is better, just to be safe.